torch/csrc/DataLoader.cpp

```diff
@@ -28,8 +28,14 @@
 static void HANDLER_NAME(int sig, siginfo_t *info, void *ctx)                 \
 {                                                                             \
   write(STDERR_FILENO, ERROR_MSG, sizeof(ERROR_MSG) / sizeof(char));          \
-  signal(SIGNAL, SIG_DFL);                                                    \
-  raise(SIGNAL);                                                              \
+  struct sigaction sa;                                                        \
+  sa.sa_handler = SIG_DFL;                                                    \
+  sa.sa_flags = 0;                                                            \
+  if (sigemptyset(&sa.sa_mask) != 0 || sigaction(SIGNAL, &sa, NULL) != 0) {   \
+    _exit(EXIT_FAILURE);                                                      \
+  } else {                                                                    \
+    raise(SIGNAL);                                                            \
```

Comment:
Shouldn't you get the signal code from `info`? I didn't even know `SIGNAL` existed

Reply:
`SIGNAL` is an argument of the macro. Both `sig` and `info` contain the same signal code, but those are generated at runtime when signal happens. So all three things are equivalent.